elastic/elastic-agent
# Use only fs.NewDownloader for file://.

## What does this PR do?



Changes the upgrade artifacts fetcher to only use the `fs.NewDownloader` when a `--source-uri` is passed with `file://` prefix.

## Why is it important?



Ensures that when a `file://` source URI is provided that a file will never be downloaded from the artifacts API. Upgrade requested to use a local file it should only look there.

## Checklist



- [x] I have added an integration test or an E2E test **(already used in tests)**

## How to test this PR locally



```
$ elastic-agent upgrade --skip-verify --source-uri file:///home/doesnt-exist 8.12.0-SNAPSHOT
Error: Failed trigger upgrade of daemon: failed download of agent binary: unable to download package: package '/home/ubuntu/elastic-agent-8.12.0-SNAPSHOT-linux-arm64.tar.gz' not found: open /home/ubuntu/elastic-agent-8.12.0-SNAPSHOT-linux-arm64.tar.gz: permission denied
For help, please see our troubleshooting guide at https://www.elastic.co/guide/en/fleet/8.12/fleet-troubleshooting.html
```
